source/Calamari/Scripts/Octopus.Features.WindowsService_AfterPreDeploy.ps1

```diff
@@ -34,11 +34,14 @@ $service = Get-Service $serviceName -ErrorAction SilentlyContinue
 if (!$service)
 {
     Write-Host "The $serviceName service does not exist yet"
+    Set-OctopusVariable -name "Octopus.Action.WindowsService.Status" -value "Stopped"
 }
 else
 {
     Write-Host "The $serviceName service already exists; it will be stopped"
     Write-Host "Stopping the $serviceName service"
+    
+    Set-OctopusVariable -name "Octopus.Action.WindowsService.Status" -value $service.Status
```

Comment:
Are you expecting the value of this variable to be `Stopped`? The .NET Service class is a bit interesting in that it loads and caches the `Status` of the service - it doesn't query the status "live". https://docs.microsoft.com/en-us/dotnet/api/system.serviceprocess.servicecontroller.refresh?view=netframework-4.7.2#System_ServiceProcess_ServiceController_Refresh

I could be wrong here and perhaps PowerShell's implementation or wrapper plays nicely.

Reply:
The e2e tests in https://github.com/OctopusDeploy/OctopusDeploy/pull/3492 (and some manual testing) report the values that I would expect, even with a service that has been started and stopped in sequence.